src/Compilers/Server/ServerShared/ServerDispatcher.cs

```diff
@@ -127,7 +128,14 @@ public void ListenAndDispatchConnections(TimeSpan? keepAlive, CancellationToken
 
             try
             {
-                Task.WaitAll(connectionList.ToArray());
+                if (connectionList.Count > 0)
```

Comment:
@agocke no.  Initially I did this to facilitate debugging some test issues I had.  I kept it because it seemed to make the intent clearer.